File: py_wip/src/liquid.py
class Liquid():
    name: str
    level: float

    def __init__(self, name: str, level: float) -> None:
        self.name = name
        self.level = level

    def __repr__(self) -> str:
        return self.__str__()

    def __str__(self) -> str:
        return f"Liq[{self.name}:{self.level}]"

    def __mod__(self, perc: float) -> "Liquid":
        source_level = self.level
        self.level = self.level - (source_level * perc)
        return Liquid(self.name, source_level * perc)

    def __truediv__(self, num: float) -> "Liquid":
        source_level = self.level
        self.level /= num
        return Liquid(self.name, source_level - self.level)

File: CSharp_wip/Source/Liquid.cs
namespace Source
{
	public class Liquid
	{
		public string name;
		public double level;

		public Liquid(string name, double level)
		{
			this.name = name;
			this.level = level;
		}

		public override string ToString()
		{
			return $"Liq[{name}:{level}]";
		}

		public static Liquid operator %(Liquid liquid, double perc)
		{
			var source_level = liquid.level;
			liquid.level -= source_level * perc;
			return new Liquid(liquid.name, source_level * perc);
		}

		public static Liquid operator /(Liquid liquid, double num)
		{
			var source_level = liquid.level;
			liquid.level /= num;
			return new Liquid(liquid.name, source_level - liquid.level);
		}
	}
}